api/src/test/java/edu/wpi/first/shuffleboard/api/util/LazyInitTest.java

```diff
@@ -0,0 +1,39 @@
+package edu.wpi.first.shuffleboard.api.util;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class LazyInitTest {
+
+  @Test
+  public void testNullInitializer() {
+    assertThrows(NullPointerException.class, () -> LazyInit.of(null), "Null initializer should throw");
+  }
+
+  @Test
+  public void testCachedValue() {
+    LazyInit<Object> lazyInit = LazyInit.of(Object::new);
+    Object value = lazyInit.get();
+    assertSame(value, lazyInit.get(), "The initial value should have been cached");
+  }
+
+  @Test
+  public void testInitializerThrows() {
+    LazyInit<Object> throwing = LazyInit.of(() -> {
+      throw new UnsupportedOperationException();
+    });
+    try {
+      throwing.get();
+      fail("No exception was thrown");
+    } catch (RuntimeException e) {
+      assertNotNull(e.getCause());
+      assertEquals(UnsupportedOperationException.class, e.getCause().getClass());
+    }
```

Comment:
Is it possible to use an `assertThrows` here?

Reply:
No, because the underlying exception is wrapped. Hence the check against the cause instead of the exception itself

Reply:
assertThrows returns the exception, so you should be able to grab it.  tldr: I think `fail` is ugly here and am trying to get rid of it.

```Java
@Test
void exceptionTesting() {
    Throwable exception = assertThrows(IllegalArgumentException.class, () -> {
        throw new IllegalArgumentException("a message");
    });
    assertEquals("a message", exception.getMessage());
}
```